To: vim_dev@googlegroups.com
Subject: Patch 7.3.270
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.270
Problem:    Illegal memory access.
Solution:   Swap conditions. (Dominique Pelle)
Files:	    src/ops.c


*** ../vim-7.3.269/src/ops.c	2011-07-15 17:51:30.000000000 +0200
--- src/ops.c	2011-08-10 12:07:22.000000000 +0200
***************
*** 6311,6317 ****
      *wc += words;
  
      /* Add eol_size if the end of line was reached before hitting limit. */
!     if (line[i] == NUL && i < limit)
      {
  	i += eol_size;
  	chars += eol_size;
--- 6311,6317 ----
      *wc += words;
  
      /* Add eol_size if the end of line was reached before hitting limit. */
!     if (i < limit && line[i] == NUL)
      {
  	i += eol_size;
  	chars += eol_size;
*** ../vim-7.3.269/src/version.c	2011-08-04 20:31:50.000000000 +0200
--- src/version.c	2011-08-10 12:09:30.000000000 +0200
***************
*** 711,712 ****
--- 711,714 ----
  {   /* Add new patch number below this line */
+ /**/
+     270,
  /**/

-- 
An indication you must be a manager:
You give constructive feedback to your dog.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
